include/lsst/afw/image/VisitInfo.h

```diff
@@ -0,0 +1,217 @@
+// -*- LSST-C++ -*- // fixed format comment for emacs
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_IMAGE_VISITINFO_H_INCLUDED
+#define LSST_AFW_IMAGE_VISITINFO_H_INCLUDED
+
+#include <cmath>
+#include <limits>
+
+#include "lsst/base.h"
+#include "lsst/daf/base.h"
+#include "lsst/afw/coord/Observatory.h"
+#include "lsst/afw/coord/Coord.h"
+#include "lsst/afw/coord/Weather.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/table/misc.h"  // for RecordId
+#include "lsst/afw/table/io/Persistable.h"
+
+namespace lsst { namespace afw { namespace image {
+
+/// Type of rotation
+enum class RotType {
+    UNKNOWN,    ///< Rotation angle is unknown. Note: if there is no instrument rotator then it is better
+                ///< to compute SKY or HORIZON and use that rotation type rather than specify UNKNOWN.
+    SKY,        ///< Orientation of E,N with respected to detector X,Y;
+                ///< X is flipped, if necessary, to match the handedness of E,N.
+    HORIZON,    ///< orientation of Az/Alt with respect to detector X,Y;
+                ///< X is flipped, if necessary, to match the handedness of Az,Alt.
+    MOUNT       ///< The position sent to the instrument rotator; the details depend on the rotator.
+};
+
+/**
+ * Information about a single exposure of an imaging camera.
+ *
+ * Includes exposure duration and date, and telescope pointing and orientation.
+ *
+ * All information is for the middle of the exposure
+ * and at the boresight (center of the focal plane).
+ * Thus for a mosaic camera VisitInfo is the same for all detectors in the mosaic.
+ *
+ * VisitInfo is immutable.
+ */
+class VisitInfo : public table::io::PersistableFacade<VisitInfo>, public table::io::Persistable {
+public:
+
+    /**
+     * Construct a VisitInfo
+     *
+     * @param[in] exposureId  exposure ID
+     * @param[in] exposureTime  exposure duration (sec)
+     * @param[in] date  TAI (international atomic time) MJD date at middle of exposure
+     * @param[in] ut1  UT1 (universal time) MJD date at middle of exposure
```

Comment:
No. UT1 is not a uniform time and DateTime cannot handle it, which is why UT1 is a float instead of a DateTime. Computing UT1 requires earth orientation predictions for the UTC date in question, e.g. as obtained from IERS Bulletin B. This information is a pain to access after the fact, and is best provided at the time the observations are taken (since telescope control systems require the information in order to track accurately).

Note: it is possible to compute earth rotation angle from UT1, but not the other way around. Since a header may include one or the other (typically neither, alas) it's safer to just have slots and provide the function elsewhere, e.g. as a method on MakeRawVisitInfo (I probably should have added that, but haven't need it yet).